### PR TITLE
Cleaned up code base and updated version of the dependencies

### PR DESCRIPTION
**Pull request by zvavybir**
This repository didn't had updates for a long time (five years) and therefor doesn't follow the current best-practices and dependency versions anymore.  This PR tries to fix that.

### CODE REVIEW

**Comment by est31** [comment]
Yeah I don't use it any more. Can you undo the formatting changes? Outside of that it's fine.

**Comment by est31** [comment]
Can you remove the commits that call rustfmt entirely from the history?

**Comment by zvavybir** [comment]
It looks to me like I can't just operate this commit out of there, because formatting does so much.  I hope that just switching to tabs is enough, otherwise I probably have to redo (more or less) everything (although this time it shouldn't take that long).

**Comment by zvavybir** [comment]
OK, I redid the changes without the formatting, it should be good now.

File: src/lib.rs [hunk 1]
```diff
@@ -289,27 +296,30 @@ pub fn read_format<'a, T :io::Read + io::Seek + 'a>(rdr :T)
...
 
 		// Now seek to right before the end to get the last packets of the content.
 		// 200 kb are just a guessed number and they might be totally wrong.
 		// Can this guess be improved??
-		try!(seek_before_end(&mut pck_rdr, 200 * 1024));
+		seek_before_end(&mut pck_rdr, 200 * 1024)?;
 
+		// We have to look into how to fix this, but it isn't of
+		// immediate importance.
+		#[warn(clippy::never_loop)]
```
Comment (est31) [inline src/lib.rs]: Can you change this to allow?

**Comment by est31** [comment]
Another thing I'm wondering about, why are you changing field access to function access?

**Review by est31 — CHANGES_REQUESTED** [review]

**Comment by est31** [comment]
> Another thing I'm wondering about, why are you changing field access to function access?

Oh that's because of ogg api changes, my bad sorry.

**Review by est31 — APPROVED** [review]
Thanks!

**Pull request merged**